enumerables.rb

```diff
@@ -0,0 +1,216 @@
+    true
+  end
+
+  def my_any?(*arg, &block)
+    if arg.length == 1
+      i = 0
+      if arg[0].is_a? Class
+        while i < length
+          return true if self[i].is_a? arg[0]
+
+          i += 1
+        end
+        false
+      else
+        i = 0
+        while i < length
+          return true if self[i] == arg
+
+          i += 1
+        end
+      end
+      false
+    end
+    block = lambda { |obj| obj } unless block_given?
+
+    i = 0
+    while i < length
+      return true if block.call(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?(*arg, &block)
+    !my_any?(*arg, &block)
+  end
+
+  def my_count(*arg, &block)
+    count = 0
+    if arg.length == 1
+      i = 0
+      while i < length
+        count += 1 if self[i] == arg[0]
+        i += 1
+      end
+      count
+    end
+    block = lambda { |obj| obj } unless block_given?
+
+    i = 0
+    while i < length
+      count += 1 if block.call(self[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_map
+    res = []
+    if is_a? Range
+      to_a.my_each { |elt| res.push(yield(elt)) }
+      res
+    end
+    return unless is_a? Array
+
+    return to_enum unless block_given?
+
+    my_each { |elt| res.push(yield(elt)) }
+    res
+  end
+
+  def my_inject(*arg)
+    if arg.length > 2
+      return ArgumentError.new "wrong number of arguments (given #{arg.length}, expected 0..2)"
+    end
+
+    return 'Error' unless (is_a? Array) || (is_a? Range)
+
+    result = 0
+    if !block_given?
+
+      return if is_a? Range
+
+      if arg.length == 1
+        return unless arg[0].is_a? Symbol # Check if it's only a symbol
+
+        our_sym = arg[0].to_s[0]
+        result = self[0]
+        i = 1
+        while i < length
+          result = result.send(our_sym, self[i])
+          i += 1
+        end
+        result
+      end
+      if arg.length == 2
+        our_sym = arg[1].to_s[0]
+        result = arg[0]
+        i = 0
+        while i < length
+          result = result.send(our_sym, self[i])
+          i += 1
+        end
+        result
+      end
+    elsif block_given?
+      if (arg.length == 1) && (arg[0].is_a? Integer)
+        if is_a? Range
+          result = to_a
+          acc = result[0]
+          i = 1
+          while i < result.length
+            acc = yield(acc, result[i])
+            i += 1
+          end
+          acc
+        end
+        result = arg[0]
+        my_each { |el| result = yield(result, el) }
+        result
+      end
+
+      if arg.length.zero?
+        if is_a? Range
+          result = to_a
+          acc = result[0]
+          i = 1
+          while i < result.length
+            acc = yield(acc, result[i])
+            i += 1
+          end
+          acc
+        elsif is_a? Array
+          result = self[0]
+          i = 1
+          while i < length
+            result = yield(result, self[i])
+            i += 1
+          end
+          result
+        end
+      end
+    end
+  end
+end
```

Comment:
## Return Local Jump Error
- When there's no block given should return Local Jump Error

```ruby
p [1, 3, 78, 90, 3].inject
# => expected: no block given (LocalJumpError)
p [1, 3, 78, 90, 3].my_inject
# => got: nil
```

## Specified a symbol
-  When a symbol is specified without an initail value combines each element of the collection by applying the symbol
```ruby
p [1, 3, 78, 90, 3].inject(:+)
# => expected: 175
p [1, 3, 78, 90, 3].my_inject(:+)
# => got: nil
```


## Symbol with an initial value with a range new class
- When a symbol is specified with an initial value combines each element of the collection by applying the symbol

```ruby
p Range.new(1, 10).inject(2, :*)
# => expected: 7257600
p Range.new(1, 10).my_inject(2, :*)
# => got: nil
```